crates/core/src/repository.rs

```diff
@@ -863,12 +847,33 @@ impl<P, S: Open> Repository<P, S> {
 
     /// Get the repository configuration
     pub fn config(&self) -> &ConfigFile {
-        self.status.config()
+        &self.open_status().config
     }
 
     // TODO: add documentation!
     pub(crate) fn dbe(&self) -> &DecryptBackend<Key> {
-        self.status.dbe()
+        &self.open_status().dbe
+    }
+
+    /// Get the [`KeyId`] of the key used to open the repository
+    pub fn key_id(&self) -> &KeyId {
+        &self.open_status().key_id
+    }
+
+    /// Delete the key with id starting with the given string from the repository.
+    ///
+    /// # Errors
+    ///
+    /// * If the key could not be removed.
+    pub fn delete_key(&self, id: &str) -> RusticResult<()> {
```

Comment:
Two things:

- what happens in case two keys have the same starting strings?
- there is no safe-guarding in place, before deleting the keys (although that might fit better into `rustic-rs`

Reply:
`find_id` returns an error if no unique id can be found.
The safeguard is the check against `key_id()` - which is the key used to open the repo.